### PR TITLE
Adds Bananium

### PR DESCRIPTION
**Pull request by brainfood1183**
Adds bananium resource as a fully fleshed out mechanic.

untitled-205

https://user-images.githubusercontent.com/113240905/233846505-29be6236-21fe-4efc-8430-92a31a5205da.mp4

Bananium

- Added bananium rock
- Added bananium ore
- Added bananium material (with stack)

- Bananium ore can be mined from asteroid and mountain rocks (quite rare) and from bananium rocks.
- Bananium ore can be processed in the ore processor giving bananium.
- Bananium is a resource, but it can also be eaten. Bananium tastes like banana and contains the Honk virus.
- Bananium when eaten leaves behind a bananium peel
- Bananium peels are slippery (between soap and syndicate soap level).
- Bananium can be juiced in a grinder.
- Bananium can be used to craft bananium structures and items.

Bananium crafting

- Added bananium clown statue
- Added bananium door
- Added bananium wall
- Added bananium horn
- Added bananium floor
- Added bananium altar
- Resprited clown wall and fixed it up so it no longer has tileing glitches.

- With bananium you can craft bananium floors (cost 2 bananium per floor). Bananium floors work like a catwalk except they squeak when you walk on them and they are slippery. Bananium floors can be deconstructed.
- With bananium you can craft bananium walls (cost 2 bananium per wall). Bananium walls have no unqie properties other than being yellow.
- With bananium you can craft bananium doors (cost 4 bananium per door). Bananium doors honk when they are interacted with.
- With bananium you can craft a bananium clown statue (costs 8 bananium per statue). Bananium statue has no unique properties.
- With bananium you can craft a bananium horn (cost 1 pipe, 4 bananium, 1 bike horn). Bananium horn sounds like an air horn.
- With bananium you can craft a bananium altar (cost 6 bananium). Use it to pray to the honkmother.

Added - Honk analyzer infector (makes testing virus much easier)

- Honk virus can only be transmitted oraly.
- Honk virus makes the infected honk, lose a small portion of health to poison and there is currently a 0.0007 chance of cluwnification.
- Honk virus can be lethal if left untreated.
- Honk virus can be cured by waiting 900 or drinking 5 bananhonk.
- Had to add disease immunity to the yml protos of all creatures with disease carrier, this includes mouse rat servant and rat king. Since honk is not contagious no need for them to be a carrier.
- Honk can be extracted from bananium by juicing it. You could potentially inject people with it.

- [x] Added screenshots and videos.

**Changelog**

:cl:
- add: Added Bananium, you can build things with it. Dont eat it!

### CODE REVIEW

**Comment by deltanedas** [comment]
HONK mech soon:tm:

**Comment by potato1234x** [comment]
finally clown virus

**Comment by deltanedas** [comment]
at first it was a joke but im actually making honk now pls instamerge thanks

File: Resources/Prototypes/Entities/Structures/Furniture/altar.yml [hunk 1]
```diff
@@ -403,3 +403,20 @@
...
+  parent: AltarBase
+  name: honkmother altar
+  description: A bananium altar dedicated to the honkmother.
+  components:
+  - type: Sprite
+    sprite: Structures/Furniture/Altars/Cults/bananium.rsi
+    state: full
+  - type: Icon
+    sprite: Structures/Furniture/Altars/Cults/bananium.rsi
+    state: full
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Structures/Furniture/altar.yml:419]: Is icon needed?

File: Resources/Prototypes/Entities/Tiles/bananium.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  - type: Clickable
+  - type: Sprite
+    netsync: false
+    sprite: Tiles/Misc/bananium.rsi
+    drawdepth: FloorTiles
+    layers:
+      - state: bananium
+  - type: Icon
+    sprite: Tiles/Misc/bananium.rsi
+    state: bananium
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Tiles/bananium.yml:16]: Don't think icon is needed.

File: Resources/Prototypes/Entities/Tiles/bananium.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            max: 1
+      - !type:DoActsBehavior
+        acts: [ "Destruction" ]
+  - type: Slippery
+    paralyzeTime: 2
+    launchForwardsMultiplier: 1.5
+  - type: StepTrigger
+    intersectRatio: 0.2
+  - type: CollisionWake
+    enabled: false
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Tiles/bananium.yml:56]: Not needed.

File: Resources/Prototypes/Entities/Tiles/bananium.yml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  - type: Physics
+    bodyType: Dynamic
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Tiles/bananium.yml:58]: This should be static because it's anchored.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Tiles/bananium.yml (hunk 2 above)
Comment (brainfood1183) [reply Resources/Prototypes/Entities/Tiles/bananium.yml:16]: isnt it needed for crafting? will test it out

File: Resources/Prototypes/Entities/Objects/Materials/materials.yml [hunk 5]
```diff
@@ -271,3 +271,59 @@
...
+  - type: Material
+    materials:
+      Bananium: 150
+  - type: Sprite
+    state: bananium_1
+  - type: Stack
+    stackType: Bananium
+    count: 10
+  - type: RadiationSource
+    intensity: 0.3
```
[the remark this answers is not in the document]
Comment (brainfood1183) [reply Resources/Prototypes/Entities/Objects/Materials/materials.yml:290]: No longer radioactive and changed sprites so no longer glows.

File: Resources/Prototypes/Entities/Objects/Materials/ore.yml [hunk 6]
```diff
@@ -166,3 +166,28 @@
...
+
+
+- type: entity
+  parent: OreBase
+  id: BananiumOre
+  name: bananium ore
+  suffix: Full
+  components:
+  - type: RadiationSource
+    intensity: 0.6
```
[the remark this answers is not in the document]
Comment (brainfood1183) [reply Resources/Prototypes/Entities/Objects/Materials/ore.yml]: as above

File: Resources/Prototypes/Entities/Structures/Decoration/statues.yml [hunk 7]
```diff
@@ -23,3 +23,18 @@
...
+  id: StatueBananiumClown
+  parent: BaseStructure
+  name: bananium savior statue
+  description: A bananium statue. It portrays the return of the savior who will rise up and lead the clowns to the great honk.
+  components:
+  - type: Sprite
+    sprite: Structures/Decoration/statues.rsi
+    state: bananium_clown
+    drawdepth: Mobs
+    offset: "0.0,0.5"
```
[the remark this answers is not in the document]
Comment (brainfood1183) [reply Resources/Prototypes/Entities/Structures/Decoration/statues.yml]: added netsync false. Not sure if you wanted this to be true.

File: Resources/Prototypes/Entities/Structures/Furniture/altar.yml (hunk 1 above)
Comment (brainfood1183) [reply Resources/Prototypes/Entities/Structures/Furniture/altar.yml]: removed icon

File: Resources/Prototypes/Entities/Tiles/bananium.yml (hunk 2 above)
Comment (brainfood1183) [reply Resources/Prototypes/Entities/Tiles/bananium.yml]: as above

File: Resources/Prototypes/Entities/Tiles/bananium.yml (hunk 3 above)
Comment (brainfood1183) [reply Resources/Prototypes/Entities/Tiles/bananium.yml]: removed.

File: Resources/Prototypes/Entities/Tiles/bananium.yml (hunk 4 above)
Comment (brainfood1183) [reply Resources/Prototypes/Entities/Tiles/bananium.yml]: it is now static.

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**